### DISCUSSION TITLE
[Doc/Question]: Difficulties interpreting the documentation connecting the Elfin-EE11

### DISCUSSION THREAD

**Discussion by valeriob** [wlcrs/huawei_solar#453] [Connecting to the inverter]
### Describe the issue

Hi,
first of all, thanks for the project !

I'm trying to complete the following setup and buying the appropriate hardware, but i'm not sure how to read the following documentation.
My setup is a SUN2000L-10KTL-M1 with the SDongleA-05 connected to my LAN for the fusionsolar connectivity, two luna2000 batteries, and i would love to add the Elfin-EE11 o the correct RS485 port.

I do not understand where should i connect the  Elfin-EE11, if on RS485A1/B1 or RS485A2/B2 ?

![image](https://github.com/wlcrs/huawei_solar/assets/1522516/c16f4114-b329-4ddf-a2ac-2d22b42d5440)

![image](https://github.com/wlcrs/huawei_solar/assets/1522516/5524bfab-2b69-4a0e-91fa-18aaa0c01a0d)

What does this do in practice ?
![image](https://github.com/wlcrs/huawei_solar/assets/1522516/d41a8ab3-d080-46f1-946e-4ec344b04cf6)

That if i have the USB SDongleA-05 you can't have any slave inverters or anything else connecte to RS485A1/B1 ? Is it a bug or documented in huawei docs somewhare ?

Thanks
Valerio

### Bescribe your Huawei Solar Setup

Inverter Type: SUN2000L-10KTL-M1
Inverter Firmware version:
SDongle present: yes
Power meter present: three phase 
Battery: LG  LUNA2000 20kWh
Battery Firmware version:

### How do you connect to the inverter?

Over serial, with a serial-to-WiFi stick

### Upload your Diagnostics File

Drag & Drop your Diagnostics File here.

### Upload your relevant debug logs

```text
nì
```

### Please confirm the following:

- [ ] The problem is still present in the latest release of this integration.
- [ ] I did not find an existing issue describing this problem.
- [ ] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by Roving-Ronin** [comment]
@valeriob There is NO bug in the integration, the recently expanded documentation was incorrect, it should have read:

"As per the following table:
 -   pins 1 & 3 are used to connect to 'downstream' devices in the chain, such as other inverters, whilst;
 -   pins 2 & 4 are used to connect to an 'upstream' device in the RS485 chain, such as a Huawei SmartLogger or Elfin EW/EE-11."

Also if you want to retain your connection to FusionSolar application for reporting / updates / management, then do not use the Elfin. Refer to this warning higher up the page:

"⚠Warning⚠: The SDongleA-05 dongle interferes with devices connected to the RS485A1 and RS485B1 pins on the inverter. If your inverter has such a dongle installed you cannot use a serial connection option. We recommend connecting to the Modbus-TCP port exposed by that dongle instead."

Best option is to assign your Dongle a static IP (reserved IP in DHCP) and connect it to your LAN/WiFi and access it via that, making sure to enable modbus access in the dongle settings (using the Installer app).
